src/ac.py

```diff
@@ -0,0 +1,212 @@
+            self.b2 = tf.get_variable("b2", [self.action_size], initializer=tf.zeros_initializer())
+
+            self.Z1 = tf.add(tf.matmul(self.state, self.W1), self.b1)  # Z1 = State * W1 +  b1
+            self.A1 = tf.nn.relu(self.Z1)  # A1 = relu(Z1)
+            self.output = tf.add(tf.matmul(self.A1, self.W2), self.b2)  # output = A1 * W2 + b2 (logits)
+
+            # Softmax probability distribution over actions
+            self.actions_distribution = tf.squeeze(tf.nn.softmax(self.output))  # actions probs = softmax(output)
+            # Loss with negative log probability
+            self.neg_log_prob = tf.nn.softmax_cross_entropy_with_logits_v2(logits=self.output,
+                                                                           labels=self.action)  # cross-entropy with actions probs.
+            self.loss = tf.reduce_mean(self.neg_log_prob * self.R_t)
+            self.optimizer = tf.train.AdamOptimizer(learning_rate=self.learning_rate).minimize(self.loss)
+
+
+def main():
+    run_with_baseline_flag = "-b" in sys.argv
+    state_size = env.observation_space.shape[0]
+    action_size = env.action_space.n
+
+    # Define hyperparameters
+    max_episodes = 5000
+    max_steps = 501
+    discount_factor = 0.99
+    policy_learning_rate = 0.0004
+    value_learning_rate = 0.05
+    render = False
+
+    # Create Logger to log scalars
+    tf_logger = TensorFlowLogger(with_baseline=run_with_baseline_flag)
+
+    # Initialize the policy network
+    tf.reset_default_graph()
+
+    # Start training the agent with REINFORCE algorithm
+    with tf.Session() as sess:
+        model = ActorCritic(state_size, action_size, policy_learning_rate, value_learning_rate, discount_factor, sess, tf_logger)
+        sess.run(tf.global_variables_initializer())
+        solved = False
+        Transition = collections.namedtuple("Transition", ["state", "action", "reward", "next_state", "done"])
+        episode_rewards = np.zeros(max_episodes)
+        average_rewards = 0.0
+        total_steps = 0
+
+        for episode in range(max_episodes):
+            steps_per_episode = 0
+            state = env.reset()
+            state = state.reshape([1, state_size])
+            episode_transitions = []
+
+            # Generate an episode, out is list of episode_transitions: state, action, reward, next_state and done.
+            for step in range(max_steps):
+                actions_distribution = sess.run(model.actor.actions_distribution, feed_dict={model.actor.state: state})
+                action = np.random.choice(np.arange(len(actions_distribution)), p=actions_distribution)
+                next_state, reward, done, _ = env.step(action)
+                next_state = next_state.reshape([1, state_size])
+
+                if render:
+                    env.render()
+
+                action_one_hot = np.zeros(action_size)
+                action_one_hot[action] = 1
+                episode_transitions.append(
```

Comment:
episode_transitions and Transition variables and irrelevant now, we need to remove them. What you think?